penguin/tools/browser_harness_tools.py

```
@@ -0,0 +1,870 @@
+                helpers = importlib.reload(helpers)
+        except ModuleNotFoundError as exc:
+            raise BrowserHarnessUnavailableError(
+                "browser-harness is not installed. It is not published on PyPI; "
+                "install `penguin-ai[browser]` for the PyDoll fallback, then "
+                "install browser-harness from a local checkout/source tree into "
+                "the same environment."
+            ) from exc
+
+        return admin, helpers
+
+    def _ensure_ready_unlocked(self):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_ensure_ready_unlocked`

(ANN202)